runelite-mixins/src/main/java/net/runelite/mixins/RSWidgetMixin.java

```diff
@@ -307,4 +307,9 @@ public Widget getChild(int index)
 		return widgets.toArray(new Widget[widgets.size()]);
 	}
 
+	@Inject
+	public boolean contains(Point point)
+	{
+		return getBounds() != null && getBounds().contains(new java.awt.Point(point.getX(), point.getY()));
+	}
```

Comment:
Wouldn't be better to just replace the custom RuneLite point with awt.Point at this point? ( 😄 ) And then just normally call getBounds().contains so this helper method would be obsolete (with simple null-check before that).

Reply:
What's the best way to do that, because that seems like a pretty big rework.

Reply:
That is true, it was just idea, and if it is big change it probably do not fits into this PR.

Reply:
I just changed EVERYTHING to use java.awt.Point and guess what, everything broke.
x and y are defined in java.awt.Point as ints, but getX and getY return doubles so i had to cast everything to an int.

Reply:
you can access the x and y directly and get them as ints without casting. So you can choose, getX/Y returns double, and .x/y returns int (the "quality"of awt API).

Reply:
I think that the way this is currently is probably fine.

Reply:
I tried it out anyway :P